settings/settingsTab.ts

```diff
@@ -0,0 +1,150 @@
+			.addToggle(toggle => toggle
+				.setValue(this.plugin.settings.keepText)
+				.onChange(async (value) => {
+					this.plugin.settings.keepText = value;
+					await this.plugin.saveSettings();
+				})
+			);
+		
+        /* === Search settings === */
+		new Setting(containerEl)
+			.setHeading()
+			.setName("Search settings");
+
+		new Setting(containerEl)
+			.setName("Parse links")
+            .setDesc("Should markdown links be displayed as just their display text in search results?")
+			.addToggle(toggle => toggle
+				.setValue(this.plugin.settings.parseLinks)
+				.onChange(async (value) => {
+					this.plugin.settings.parseLinks = value;
+					await this.plugin.saveSettings();
+				})
+			);
+        
+            new Setting(containerEl)
+                .setName("Remove block id from block content")
+                .setDesc("Should the ^block-id text be removed from the block contents in search results (it will still be displayed beneath the block contents).")
+                .addToggle(toggle => toggle
+                    .setValue(this.plugin.settings.removeIdFromContent)
+                    .onChange(async (value) => {
+                        this.plugin.settings.removeIdFromContent = value;
+                        await this.plugin.saveSettings();
+                    })
+                );
+		
+		new Setting(containerEl)
+			.setName("Use selected text as initial search")
+            .setDesc("If true, when you open the search box, the text you have selected will already be searched for.")
+			.addToggle(toggle => toggle
+				.setValue(this.plugin.settings.selectedTextAsSearch)
+				.onChange(async (value) => {
+					this.plugin.settings.selectedTextAsSearch = value;
+					await this.plugin.saveSettings();
+				})
+			);
+		
+		new Setting(containerEl)
+			.setName("Search limit")
+            .setDesc("The number of search results to display.")
+			.addSlider(slider => slider
+                .setLimits(1,50,1)
+                .setValue(this.plugin.settings.searchLimit)
+                .setDynamicTooltip()
+                .onChange(async (value) => {
+                    this.plugin.settings.searchLimit = value
+                    await this.plugin.saveSettings();
+                })
+            );
+        
+        new Setting(containerEl)
+            .setName("File name in result")
+            .setDesc("Do you want the names of the files in the search results to displayed as their/full/path.md or just as their name?")
+            .addDropdown(dropdown => dropdown
+                .addOption("base", "File name")
+                .addOption("path", "File path")
+                .setValue(this.plugin.settings.fileName)
+                .onChange(async (value) => {
+                    this.plugin.settings.fileName = value
+                    await this.plugin.saveSettings()
+                })
+            );
+		
+        /* === What to search */
```

Comment:
:pinching_hand: tiny consistency with above
```suggestion
        /* === What to search === */
```